api/fees/fees.go

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) 2025 The VeChainThor developers
+
+// Distributed under the GNU Lesser General Public License v3.0 software license, see the accompanying
+// file LICENSE or <https://www.gnu.org/licenses/lgpl-3.0.html>
+
+package fees
+
+import (
+	"math"
+	"net/http"
+	"strconv"
+
+	"github.com/gorilla/mux"
+	"github.com/pkg/errors"
+	"github.com/vechain/thor/v2/api/utils"
+	"github.com/vechain/thor/v2/bft"
+	"github.com/vechain/thor/v2/chain"
+)
+
+type Fees struct {
+	data           *FeesData
+	bft            bft.Committer
+	backtraceLimit uint32 // The max number of blocks to backtrace.
+}
+
+func New(repo *chain.Repository, bft bft.Committer, backtraceLimit uint32, fixedCacheSize uint32) *Fees {
+	return &Fees{
+		data:           newFeesData(repo, fixedCacheSize),
+		bft:            bft,
+		backtraceLimit: backtraceLimit,
+	}
+}
+
+func (f *Fees) validateGetFeesHistoryParams(req *http.Request) (uint32, *chain.BlockSummary, error) {
+	// Validate blockCount
+	blockCountParam := req.URL.Query().Get("blockCount")
+	blockCount, err := strconv.ParseUint(blockCountParam, 10, 32)
+	if err != nil {
+		return 0, nil, utils.BadRequest(errors.WithMessage(err, "invalid blockCount, it should represent an integer"))
+	}
+
+	if blockCount == 0 {
```

Comment:
Given the conversation we had offline, do you think it's necessary to also have the check of `blockCount>f.backtraceLimit {return utils.BadRequest}` ? @otherview

Reply:
No because in that case we just return what is available

Reply:
What do you mean by "available"?

Reply:
Following the other example, if we have 10 blocks, backtrace limit is 5 and we request from newestBlock 10 to blockCount 6, we return 5 blocks. That is how it works in Ethereum btw.

Reply:
For context, the cut allows for a user to get blockCount == f.backtraceLimit easily. Otherwise it would be very easy to have an off by 1 request.

Reply:
I believe it is beneficial to alert users about obvious input errors. Notifying them about an 'out of range blockCount' can provide helpful guidance during the development stage.

Reply:
I was considering `blockCount>f.backtraceLimit` to be a user input error. And cut off should work if user input`blockID=100, blockCount=10, backtraceLimit=10` and when the request comes `bestBlock =101`, then cut off has value for user since the blockchain ticked right during the time of processing the block. @otherview @freemanzMrojo wdyt?